### ISSUE TITLE
Realtime output from a shell command in Jupyter notebook

### ISSUE THREAD

**Issue by bfurtwa** [jupyter/notebook#4048]
I'm telling jupyter to execute a python script:

!python build_database.py

When executed from the terminal, the python script prints the progress during execution. However, in the jupyter notebook, I get all output printed as a list of strings, AFTER the execution. Is there a way to see the output live?

**Comment by amrs-tech** [comment]
@benni93 Can you please provide any screenshots for your both cases ?

**Comment by benni93** [comment]
This is from the jupyter notebook. It is not live, the output is shown after everything is finished.

<img width="841" alt="unbenannt" src="https://user-images.githubusercontent.com/23263654/46248716-1c23bd00-c41d-11e8-98f8-b36cec274010.PNG">

This is from the shell script. The output is live, I see every iteration.

<img width="960" alt="unbenannt" src="https://user-images.githubusercontent.com/23263654/46248741-863c6200-c41d-11e8-9ab6-35fe278da726.PNG">

**Comment by amrs-tech** [comment]
Have you tried running in the notebook as follows ?
%run build_database.py

**Comment by benni93** [comment]
Yes, but I can´t run the script in the jupyter notebook, since it uses asyncio. (https://github.com/jupyter/notebook/issues/3397)

**Comment by amrs-tech** [comment]
I too a newcomer as a contributor. I'll try to help you. I'll check for any solution to your problem. But I think I might take some time.

**Comment by bollwyvl** [comment]
If async was holding you back, turns out you're in luck:  ipython 7 with
ipykernel 5 from pip or conda-forge have got you covered:

https://blog.jupyter.org/ipython-7-0-async-repl-a35ce050f7f7

If you do run into issues, make sure to raise them over in @ipython/ipython
!

**Comment by krassowski** [comment]
Closing as this is a solved problem with IPython 8.32:

https://github.com/user-attachments/assets/34b32392-7b8e-4d4f-a624-ba27592ff2ed

If that does not work for you it might be because of Python buffering see https://docs.python.org/3/using/cmdline.html#envvar-PYTHONUNBUFFERED.

In future such issues would be best opened on repository of respective kernels or underlying implementations.
